dashboard/config/routes.rb

```diff
@@ -361,6 +361,8 @@
     end
   end
 
+  get '/programming_environments/get_summary_by_name/:name', to: 'programming_environments#get_summary_by_name'
```

Comment:
I think you can add this as a `member` in the `programming_environment` block above. Something like:
```
resources :programming_environments, only: [:index, :new, :create, :edit, :update, :show, :destroy], param: 'name' do
  member do
    get :get_summary_by_name
  end
  resources :programming_expressions
  ...
  end
end
```
It would be nice to keep all the routes together but if it doesn't work then it's okay here

Reply:
thanks that's better!